components/notification/notification.service.ts

```diff
@@ -25,39 +27,39 @@ export class NzNotificationService extends NzMNService<NzNotificationContainerCo
 
   success(
     title: string | TemplateRef<void>,
-    content: string | TemplateRef<void>,
+    content: string | TemplateRef<{ $implicit: NzNotificationComponent; data: NzSafeAny }>,
```

Comment:
As you said, we need compatibility :)

```suggestion
    content: string | TemplateRef<void | { $implicit: NzNotificationComponent; data: NzSafeAny }>,
```